### DISCUSSION TITLE
Optional Tuples break typer

### DISCUSSION THREAD

**Discussion by multimeric** [fastapi/typer#667] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Optional, Tuple
import typer

def main(arg: Optional[Tuple[int]] = None):
    pass

typer.run(main)

```
```
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /opt/anaconda3/envs/napari-env/lib/python3.9/site-packages/typer/main.py:650 in                  │
│ internal_convertor                                                                               │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
TypeError: 'NoneType' object is not iterable
```

### Description

Typer seems to hit this error when it encounters `Optional[Tuple[...]]`. It's readily reproducible.

### Operating System

macOS

### Operating System Details

Monterey, 12.6

### Typer Version

0.9.0

### Python Version

3.9.16

**Comment by Asthestarsfalll** [comment]
Same problem

**Comment by svlandeg** [comment]
Hi!

To configure an option that expects [multiple values](https://typer.tiangolo.com/tutorial/multiple-values/multiple-options/), you can use a `List`:

```
def main(arg: Optional[List[int]] = None):
    pass
```

or with `Annotated`:
```
def main(arg: Annotated[Optional[List[int]], typer.Option()] = None):
    pass
```

**Reply by multimeric** [reply]
A list accepts an arbitrary number of values, whereas a type such as `Optional[Tuple[int, int]]` means "allow either `None`, or two integers, but not one integer or three etc". So your solution doesn't solve my use case.

**Reply by svlandeg** [reply]
That's true, and it does look like that's not quite supported yet, and we'll look into it.

**Reply by Asthestarsfalll** [reply, marked as answer]
I submit a [PR](https://github.com/tiangolo/typer/pull/757) to support this
